testsuite/tests/comprehensions/array_comprehensions_pure.ml

```diff
@@ -409,3 +409,42 @@ Line 1, characters 23-24:
 Warning 18 [not-principal]: this type-based constructor disambiguation is not principal.
 - : M.t array = [|M.B; M.A; M.A; M.A|]
 |}];;
+
```

Comment:
You went down a bit too far!  This is the "Test that types are being propagated inwards correctly" section, which is about constructor disambiguation.  You'll want to bring it up above that back into the "Errors" section, and add a leading comment line saying what's in these tests: `(* Module patterns are disallowed (for now) *)`, for instance.  Maybe at the end of the section, maybe right after "No duplicating variables in a for-and clause" (which see for formatting).

Reply:
Actually resolved now :)